python/examples/core/transform.py

```diff
@@ -209,6 +239,25 @@ def print_ir(self,
         transforms.append(PrintIR(name=str(t)))
     return TransformationList(transforms=transforms)
 
+  def save_ir(self,
+              file_name,
+              after_all: bool = False,
+              llvm: bool = False) -> TransformationList:
+    """Return a new transformation list that save the IR at the given points."""
+    transforms = []
+    id = 0
+    for t in self.transforms:
+      if (len(transforms) >= 2 and
+          type(transforms[-2]).__name__ == type(t).__name__):
```

Comment:
Do you see value in this type of granularity?
I would have thought that set by step saving and incrementing would be simpler and nicer.

I.e. atm you'd get something like:
```
foo.transA.0
foo.transA.1
foo.transB.0
foo.transB.1
```
which does not tell you how A and B are interleaved.

How about 
```
foo.1.transA.
foo.2.transB
foo.3.transA
foo.4.transB
```

instead ?

Reply:
coming back to this, I don't think this has been addressed ?

Reply:
No, sorry, I didn't notice @ftynse comment. I am going to push it now

Reply:
Done :)